webapp/src/ts/modules/analytics/analytics.component.ts

```diff
@@ -44,30 +44,38 @@ export class AnalyticsComponent implements OnInit, OnDestroy {
   }
 
   private subscribeToStore() {
-    const selectorsSubscription = combineLatest(
-      this.store.select(Selectors.getAnalyticsModules),
-    )
-      .subscribe(([
-        analyticsModules = [],
-      ]) => {
+    const selectorsSubscription = combineLatest(this.store.select(Selectors.getAnalyticsModules))
+      .subscribe(([ analyticsModules = [] ]) => {
         this.analyticsModules = analyticsModules;
       });
     this.subscriptions.add(selectorsSubscription);
   }
 
+  private subscribeRouterNavigation() {
+    const subscription = this.router.events
+      .pipe(filter(event => event instanceof NavigationEnd))
+      .subscribe(() => this.redirectToModule(this.analyticsModules));
+    this.subscriptions.add(subscription);
+  }
+
+  private redirectToModule(analyticsModules) {
+    if (!analyticsModules) {
+      return;
+    }
+
+    const isAnalyticsTab = this.route.snapshot.firstChild?.data?.tab === 'analytics';
+
+    if (isAnalyticsTab && analyticsModules.length === 1) {
+      this.router.navigate(analyticsModules[0].route);
+    }
+  }
+
   private getAnalyticsModules() {
     return this.analyticsModulesService
       .get()
-      .then((modules) => {
+      .then(modules => {
         this.analyticsActions.setAnalyticsModules(modules);
-        if (this.route.snapshot.firstChild?.data?.tab === 'analytics') {
-          if (modules.length === 1) {
-            return this.router.navigate(modules[0].route);
-          }
-        } else {
-          const selectedAnalytics = _find(modules, { id: this.route.snapshot?.firstChild?.data?.moduleId });
-          this.analyticsActions.setSelectedAnalytics(selectedAnalytics);
-        }
+        this.redirectToModule(modules);
```

Comment:
I think it's a little risky to pass `modules` here as a parameter, given that we use the value from the store in other places. 
It's risky because we don't know if the reducer doesn't change the values that we pass to it before storing them, and we could end up with using different values when loading vs when inner navigation happens.

Reply:
Using store to access modules